### PR TITLE
refactor(stylelint): refactor to and enforce kebab-casing for Sass utilities

### PR DESCRIPTION
**Pull request by Elijbet**
**Related Issue:** #7365

## Summary
Refactor sass names to kebab-case.

### CODE REVIEW

File: packages/calcite-components/.stylelintrc.json [hunk 1]
```diff
@@ -40,6 +40,30 @@
...
+    "scss/at-function-pattern": [
+      "^([a-z][a-z0-9]*)(-[a-z0-9]+)*$",
+      {
+        "message": "Expected class selector to be kebab-cased."
+      }
+    ],
+    "scss/dollar-variable-pattern": [
+      "^([a-z][a-z0-9]*)(-[a-z0-9]+)*$",
+      {
+        "message": "Expected class selector to be kebab-cased."
```
Comment (jcfranco) [inline packages/calcite-components/.stylelintrc.json:53]: Can you set the `severity` to `error`? Applies to all.

File: packages/calcite-components/.stylelintrc.json [hunk 2]
```diff
@@ -40,6 +40,30 @@
       {
         "ignoreTypes": ["/^calcite-/"]
       }
+    ],
+    "scss/at-function-pattern": [
+      "^([a-z][a-z0-9]*)(-[a-z0-9]+)*$",
+      {
+        "message": "Expected class selector to be kebab-cased."
```
Comment (jcfranco) [inline packages/calcite-components/.stylelintrc.json:47]: All messages say `Expected class selector`, which is not accurate. Can we simplify these to "<target> should be kebab-cased"? 

The message does appear next to the name, so we could also use `Name should be kebab-cased`. I'll defer to you on this.

**Review by jcfranco — CHANGES_REQUESTED** [review]
Looking good!

**Review by jcfranco — APPROVED** [review]
🎉

Before merging, can you revisit the PR title to convey that this enforces kebab-casing for Sass utils?

**Review by driskull — APPROVED** [review]
LGTM!! 👍

**Pull request merged**